src/MainWindow.vala

```diff
@@ -76,6 +76,16 @@ namespace PantheonCalculator {
             });
         }
 
+        public void undo () {
+            unowned List<History?> prev_entry = history.last ();
+            uint hist_length = history.length () - 1;
+            unowned List<History?> last_el = history.nth (hist_length);
```

Comment:
It seems as though you're creating a reference to the same element twice. Isn't the element at `history.length -1` the same as `history.last`?